testcontainers/src/main/java/io/streamnative/oxia/testcontainers/OxiaContainer.java

```diff
@@ -38,18 +41,29 @@ public class OxiaContainer extends GenericContainer<OxiaContainer> {
             DockerImageName.parse("streamnative/oxia:main");
 
     public OxiaContainer(@NonNull DockerImageName imageName) {
-        this(imageName, DEFAULT_SHARDS);
+        this(imageName, DEFAULT_SHARDS, false);
     }
 
+    public OxiaContainer(@NonNull DockerImageName imageName, int shards) {
+        this(imageName, shards, false);
+    }
+
+    @SneakyThrows
     @SuppressWarnings("resource")
-    OxiaContainer(@NonNull DockerImageName imageName, int shards) {
+    public OxiaContainer(@NonNull DockerImageName imageName, int shards, boolean fixedServicePort) {
         super(imageName);
         this.imageName = imageName;
         this.shards = shards;
         if (shards <= 0) {
             throw new IllegalArgumentException("shards must be greater than zero");
         }
-        addExposedPorts(OXIA_PORT, METRICS_PORT);
+        if (fixedServicePort) {
+            int freePort = findFreePort();
+            addFixedExposedPort(freePort, OXIA_PORT);
```

Comment:
Why exposing a fixed port? There is already a way to get the port with random number and use it outside the container (eg: `getMappedPort(OXIA_PORT)`)

Reply:
because I wanna testing the client behaviour when server restart. existing mechanism will use different ports after restart the container. that will make the client can't reonnect to the server.